awswrangler/data_api/redshift.py

```diff
@@ -110,26 +112,25 @@ def _execute_statement(
 
         self._validate_redshift_target()
         self._validate_auth_method()
-        credentials = {}
+        args = {}
         if self.secret_arn:
-            credentials = {"SecretArn": self.secret_arn}
-        elif self.db_user:
-            credentials = {"DbUser": self.db_user}
+            args["SecretArn"] = self.secret_arn
+        if self.db_user:
+            args["DbUser"] = self.db_user
 
         if database is None:
             database = self.database
 
         if self.cluster_id:
-            redshift_target = {"ClusterIdentifier": self.cluster_id}
-        elif self.workgroup_name:
-            redshift_target = {"WorkgroupName": self.workgroup_name}
+            args["ClusterIdentifier"] = self.cluster_id
+        if self.workgroup_name:
+            args["WorkgroupName"] = self.workgroup_name
```

Comment:
What happens if I try to pass both `secret_arn` and `db_user` or both `cluster_id` and `workgroup_name`?

Reply:
Boto validation will pick that up and come back with `botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the ExecuteStatement operation: Cannot apply SecretManager ARN and DB User in parallel.`

Reply:
Added a check to validate that and throw a nicer error message